### PR TITLE
boards/native/doc: Update architecture description

### PR DESCRIPTION
**Pull request by aescolar**
The architecture description was quite out of date in some parts, specially those which changed with the introduction of the native simulator.
Let's update it.

Link to rendered docs (when completed)
https://builds.zephyrproject.io/zephyr/pr/78993/docs/boards/native/doc/arch_soc.html

### CODE REVIEW

File: boards/native/doc/arch_soc.rst [hunk 1]
```diff
@@ -304,138 +304,185 @@ side-effects.
...
+together into one executable with more elaborate HW models of those SOCs.
+
+Native simulator runner context and the embedded context
+========================================================
+
+It is worth noting that the embedded SW library is first pre-linked. That is that all symbols which
+can be resolved inside that library will be resolved. And that, after, all these library symbols,
+except a selected few marked with an special annotation, will be hidden from further linking.
+In this way, the runner link stage will not link to or conflict with any of these hidden symbols,
+neither from the runner itself or from other embedded CPUs libraries.
```
Comment (jori-nordic) [inline boards/native/doc/arch_soc.rst:368]: what do you mean exactly by "embedded CPU libraries"? The pre-linked lib that has the embedded context, or some other library

File: boards/native/doc/arch_soc.rst [hunk 2]
```diff
@@ -304,138 +304,185 @@ side-effects.
...
+Another is the runner context, which is always built with the host C library and which has very
+limited visibility into the embedded context.
+
+From the embedded context we can easily call into the runner context: All runner context symbols
+will be linkable in the final link stage unless another embedded symbol with the same name was
+already linked to it in the first pass.
+But from the runner context only the symbols from the embedded context annotated with the
+``NATIVE_SIMULATOR_IF`` macro will be linkable.
+
+From Zephyr's build system it is possible to request a file to be build in the runner context by
```
Comment (jori-nordic) [inline boards/native/doc/arch_soc.rst:388]: to be built

File: boards/native/doc/arch_soc.rst [hunk 3]
```diff
@@ -304,138 +304,185 @@ side-effects.
...
-In this architecture each Zephyr thread is mapped to one POSIX pthread.
-The POSIX architecture emulates a single threaded CPU/MCU by only allowing
-one SW thread to execute at a time, as commanded by the Zephyr kernel.
-Whenever the Zephyr kernel desires to context switch two threads,
-the POSIX arch blocks and unblocks the corresponding pthreads.
+The POSIX architecture is mainly responsible for two things:
+
+* Set up the Zephyr build to produce an static library for the host architecture to be later
+  used with the native simulator build and linked with the native simulator runner.
+* Provides a thin adaptation between the API the Zephyr kernel expects from an architecture
```
Comment (jori-nordic) [inline boards/native/doc/arch_soc.rst:404]: "Provide" to match "Set up"

**Review by jori-nordic — DISMISSED** [review]

**Review by theob-pro — DISMISSED** [review]

File: boards/native/doc/arch_soc.rst (hunk 1 above)
Comment (aescolar) [reply boards/native/doc/arch_soc.rst:368]: I mean "The pre-linked lib that has the embedded context"  :)

**Comment by aescolar** [comment]
Thanks for the reviews so far @jori-nordic & @theob-pro . I updated it based on @jori-nordic feedback.

**Review by theob-pro — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**